runtime/service/src/test/java/org/apache/polaris/service/events/jsonEventListener/aws/cloudwatch/AwsCloudWatchEventListenerTest.java

```diff
@@ -0,0 +1,311 @@
+            return client;
+          }
+        };
+
+    // Set up call context and security context
+    CallContext callContext = Mockito.mock(CallContext.class);
+    PolarisCallContext polarisCallContext = Mockito.mock(PolarisCallContext.class);
+    RealmContext realmContext = Mockito.mock(RealmContext.class);
+    SecurityContext securityContext = Mockito.mock(SecurityContext.class);
+    Principal principal = Mockito.mock(Principal.class);
+    when(callContext.getRealmContext()).thenReturn(realmContext);
+    when(callContext.getPolarisCallContext()).thenReturn(polarisCallContext);
+    when(realmContext.getRealmIdentifier()).thenReturn(REALM);
+    when(securityContext.getUserPrincipal()).thenReturn(principal);
+    when(principal.getName()).thenReturn(TEST_USER);
+    listener.callContext = callContext;
+    listener.securityContext = securityContext;
+
+    return listener;
+  }
+
+  @Test
+  void shouldCreateLogGroupAndStream() {
+    CloudWatchLogsAsyncClient client = createCloudWatchAsyncClient();
+    AwsCloudWatchEventListener listener = createListener(client);
+
+    // Start the listener which should create the log group and stream
+    listener.start();
+
+    try {
+      // Verify log group exists
+      DescribeLogGroupsResponse groups =
+          client
+              .describeLogGroups(
+                  DescribeLogGroupsRequest.builder().logGroupNamePrefix(LOG_GROUP).build())
+              .join();
+      assertThat(groups.logGroups())
+          .hasSize(1)
+          .first()
+          .satisfies(group -> assertThat(group.logGroupName()).isEqualTo(LOG_GROUP));
+
+      // Verify log stream exists
+      DescribeLogStreamsResponse streams =
+          client
+              .describeLogStreams(
+                  DescribeLogStreamsRequest.builder()
+                      .logGroupName(LOG_GROUP)
+                      .logStreamNamePrefix(LOG_STREAM)
+                      .build())
+              .join();
+      assertThat(streams.logStreams())
+          .hasSize(1)
+          .first()
+          .satisfies(stream -> assertThat(stream.logStreamName()).isEqualTo(LOG_STREAM));
+    } finally {
+      // Clean up
+      listener.shutdown();
+      client.close();
+    }
+  }
+
+  @Test
+  void shouldSendEventToCloudWatch() {
+    CloudWatchLogsAsyncClient client = createCloudWatchAsyncClient();
+    AwsCloudWatchEventListener listener = createListener(client);
+    listener.start();
+    try {
+      // Create and send a test event
+      TableIdentifier testTable = TableIdentifier.of("test_namespace", "test_table");
+      AfterTableRefreshedEvent event = new AfterTableRefreshedEvent(testTable);
+      listener.onAfterTableRefreshed(event);
+
+      // Verify the event was sent to CloudWatch by reading the actual logs
+      GetLogEventsResponse logEvents;
+      int eventCount = 0;
+      long startTime = System.currentTimeMillis();
+      while (eventCount == 0) {
```

Comment:
We can use [awaitility](https://github.com/awaitility/awaitility) here, it's already in the project so no big lift here

Reply:
Good call - I've updated this in the next revision.